tooling/Microsoft.VisualStudio.BlazorExtension/AutoRebuild/AutoRebuildService.cs

```diff
@@ -0,0 +1,122 @@
+        }
+
+        public void Listen()
+        {
+            AddBuildServiceNamedPipeServer();
+        }
+
+        private void AddBuildServiceNamedPipeServer()
+        {
+            Task.Factory.StartNew(async () =>
+            {
+                try
+                {
+                    var identifier = WindowsIdentity.GetCurrent().Owner;
+                    var security = new PipeSecurity();
+
+                    // Restrict access to just this account.  
+                    var rule = new PipeAccessRule(identifier, PipeAccessRights.ReadWrite | PipeAccessRights.CreateNewInstance, AccessControlType.Allow);
+                    security.AddAccessRule(rule);
+                    security.SetOwner(identifier);
+
+                    using (var serverPipe = new NamedPipeServerStream(
+                        _pipeName,
+                        PipeDirection.InOut,
+                        NamedPipeServerStream.MaxAllowedServerInstances,
+                        PipeTransmissionMode.Byte,
+                        PipeOptions.Asynchronous | PipeOptions.WriteThrough,
+                        0x10000, // 64k input buffer
+                        0x10000, // 64k output buffer
+                        security,
+                        HandleInheritability.None))
+                    {
+                        // As soon as we receive a connection, spin up another background
+                        // listener to wait for the next connection
+                        await serverPipe.WaitForConnectionAsync();
+                        AddBuildServiceNamedPipeServer();
+
+                        await HandleRequestAsync(serverPipe);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    await AttemptLogErrorAsync(
+                        $"Error in Blazor AutoRebuildService:\n{ex.Message}\n{ex.StackTrace}");
+                }
+            }, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
+        }
+
+        private async Task HandleRequestAsync(Stream stream)
+        {
+            // Protocol:
+            //   1. Send a "protocol version" number to the client
+            //   2. Receive the project path from the client
+            //      If it is the special string "abort", gracefully disconnect and end
+            //      This is to allow for mismatches between server and client protocol version
+            //   3. Receive the "if not built since" timestamp from the client
+            //   4. Perform the build, then send back the success/failure result flag
+            // Keep in sync with VSForWindowsRebuildService.cs in the Blazor.Server project
+            // In the future we may extend this to getting back build error details
+            await stream.WriteIntAsync(_protocolVersion);
+            var projectPath = await stream.ReadStringAsync();
```

Comment:
This also needs to check if the client is running as admin. That was the second link I provided.

Reply:
or rather, it should check if the client is at the same privilege level